astroquery/jplhorizons/core.py

```diff
@@ -1051,24 +1116,39 @@ def vectors_async(self, get_query_payload=False,
 
     # ---------------------------------- parser functions
 
-    def _parse_horizons(self, src):
+    def _parse_result(self, response, verbose=None):
         """
-        Routine for parsing data from JPL Horizons
+        Parse query result to a `~astropy.table.Table` object.
 
 
         Parameters
         ----------
-        self : HorizonsClass instance
-        src : list
-            raw response from server
+
+        response : `~requests.Response`
+            Response from server.
 
 
         Returns
         -------
-        data : `astropy.Table`
+
+        data : `~astropy.table.Table`
+
         """
 
-        self.raw_response = src
+        self.last_response = response
+        try:
+            response.raise_for_status()
+        except HTTPError:
+            # don't cache any HTTP errored queries (especially when the API is down!)
+            try:
+                self._last_query.remove_cache_file(self.cache_location)
+            except OSError:
+                # this is allowed: if `cache` was set to False, this
+                # won't be needed
+                pass
+            raise
+
+        self.raw_response = response.text
```

Comment:
I think we should deprecate this one as well, is it used anywhere?

Reply:
Done.